### PR TITLE
Feature #13789: improving performances.

### PR DESCRIPTION
**Pull request by SilverYoCha**
Adding a new signature to comment services in order to get comment count of all resources of a specified type hosted by a component instance.

Linked to PR https://github.com/Silverpeas/Silverpeas-Components/pull/849

### CODE REVIEW

File: core-services/comment/src/main/java/org/silverpeas/core/comment/dao/CommentDAO.java [hunk 1]
```diff
@@ -91,11 +92,22 @@ List<Comment> getAllCommentsByForeignKey(final String resourceType,
...
+  /**
+   * Gets the number of comments by resources of resource type and hosted by the specified
+   * component instance.
+   *
+   * @param resourceType type of the commented publication.
+   * @param instanceId identifier of aimed component instance.
+   * @return a map containing the number of comments by resources.
+   */
+  Map<ResourceReference, Integer> getInstanceCommentsCountByResourceRefs(final String resourceType,
+      final String instanceId);
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-services/comment/src/main/java/org/silverpeas/core/comment/dao/CommentDAO.java:109]: Signature has been renamed.

File: core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java [hunk 2]
```diff
@@ -278,6 +281,32 @@ public int getCommentsCount(Connection con, String resourceType, ResourceReferen
     return commentsCount;
   }
 
+  public Map<ResourceReference, Integer> getInstanceCommentsCountByResourceRefs(Connection con,
+      String resourceType, String instanceId) throws SQLException {
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java:285]: Signature has been renamed.

File: core-services/comment/src/main/java/org/silverpeas/core/comment/service/CommentService.java [hunk 3]
```diff
@@ -197,6 +198,17 @@ List<CommentedPublicationInfo> getMostCommentedPublicationsInfo(final String res
...
+  /**
+   * Gets the number of comments by resources of resource type and hosted by the specified
+   * component instance.
+   *
+   * @param resourceType type of the commented publication.
+   * @param instanceId identifier of aimed component instance.
+   * @return a map containing the number of comments by resources.
+   */
+  Map<ResourceReference, Integer> getInstanceCommentsCountByResourceRefs(final String resourceType,
+      final String instanceId);
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-services/comment/src/main/java/org/silverpeas/core/comment/service/CommentService.java:210]: Signature has been renamed.

File: core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmoqui) [reply core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java:285]: No. Doesn't seem to be renamed

File: core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentDAO.java [hunk 4]
```diff
@@ -139,14 +140,23 @@ public List<CommentedPublicationInfo> getMostCommentedPublications(final String
...
+    }
+  }
+
+  @Override
+  public Map<ResourceReference, Integer> getCommentCountIndexedByResource(
+      final String resourceType, final String instanceId) {
+    try (Connection con = openConnection()) {
+      final JDBCCommentRequester requester = getRequester();
+      return requester.getInstanceCommentsCountByResourceRefs(con, resourceType, instanceId);
     } catch (Exception e) {
```
Comment (mmoqui) [inline core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentDAO.java:157]: The method of the request isn't renamed here

**Review by mmoqui — CHANGES_REQUESTED** [review]

File: core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-services/comment/src/main/java/org/silverpeas/core/comment/dao/jdbc/JDBCCommentRequester.java:285]: Shit. That is a forgot.

**Review by mmoqui — APPROVED** [review]

**Pull request merged**